Add a new symbol flag to mark addresses within method bounds

dmitry-ten opened the pull request:
A new flag AddressWithinMethodBounds indicates that an address
stored in a static symbol is inside method bounds, and can be
accessed using RIP addressing during an out-of-process compilation.

dmitry-ten commented:
Related to https://github.com/eclipse-openj9/openj9/pull/13458.
@mpirvu could you review?
You can notice that the flag value is reused in other places, but so are other static symbol flag values, so it probably means they cannot be both true at the same time.

mpirvu reviewed (commented):
LGTM. I cannot tell what the best location for the new flag is (between flags and flags2), so maybe someone with more OMR experience can pitch in.

mpirvu commented:
@0xdaryl Could you please review this PR or delegate? Thanks

0xdaryl commented:
* Can you add a comment in the code documenting what the semantics of this flag are?

* Perhaps it should be called "StaticAddressWithinMethodBounds" if you are restricting this to static symbols.  There are other non-static symbols where this flag could apply.  For example, LabelSymbols, which are always within a method's bounds but are not static symbols.

* Did you verify that all the other static symbols that we create have this new flag set appropriately for that kind of symbol?

dmitry-ten commented:
Added comment and renamed the flag.

Marius and I looked at places where `NotDataAddress` flag is set for static symbols. There seem to be just 2 locations where the address is within method bounds. For other static addresses, they describe some data, so they are most likely not going to be within method bounds.

0xdaryl commented:
Jenkins build all
